### PR TITLE
Update config / code styles for 4.0

### PR DESCRIPTION
**Pull request by tractorcow**
Related https://github.com/silverstripe/silverstripe-framework/issues/7590

Fixes include:
 - support for getenv for _configure_database.php
 - coding style enhancements
 - update CI config to trusty

### CODE REVIEW

**Comment by tractorcow** [comment]
Build is broken but it's still better than the prior breaking build. :)

File: .travis.yml [hunk 1]
```diff
@@ -1,34 +1,40 @@
...
-  - printf "\n" | pecl install imagick
   - phpenv rehash
   - phpenv config-rm xdebug.ini
+  - export PATH=~/.composer/vendor/bin:$PATH
   - echo 'memory_limit = 2048M' >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
 
 # Install composer dependencies
   - composer validate
-  - composer require symfony/config:^3.2 silverstripe/framework:4.0.x-dev silverstripe/cms:4.0.x-dev silverstripe/siteconfig:4.0.x-dev silverstripe/config:1.0.x-dev silverstripe/admin:1.0.x-dev silverstripe/assets:1.0.x-dev silverstripe/versioned:1.0.x-dev --no-update
+  - composer require --prefer-dist --no-update silverstripe/recipe-cms:1.0.x-dev
```
Comment (dhensby) [inline .travis.yml]: `--prefer-dist` here does nothing

Comment (tractorcow) [reply .travis.yml]: Not all dependencies of recipe-cms are dev branches. E.g. symfony.

**Comment by tractorcow** [comment]
I've updated as much as I can... there are other issues but are pre-existing, or in framework regressions.

File: .travis.yml (hunk 1 above)
Comment (dhensby) [reply .travis.yml]: Yes, but when using `--no-update` only the composer file is modified and no installs happen. There is no meta data stored that means when you subsequently run an install that `--prefer-dist` will be used.

The next line runs the install and correctly uses the `--prefer-dist` flag

**Comment by tractorcow** [comment]
Ok, updated @dhensby

**Pull request merged**